pilot/pkg/serviceregistry/kube/client.go

```diff
@@ -70,6 +71,14 @@ func ResolveConfig(kubeconfig string) (string, error) {
 
 // CreateInterface is a helper function to create Kubernetes interface from kubeconfig file
 func CreateInterface(kubeconfig string) (kubernetes.Interface, error) {
+	if len(kubeconfig) == 0 {
+		// Avoid the confusing "Things might not work" message
+		restConfig, err := restclient.InClusterConfig()
+		if err != nil {
+			return nil, err
+		}
+		return kubernetes.NewForConfig(restConfig)
```

Comment:
? @ayj had a similar PR

Reply:
Well, happy to have his merged - it creates a lot of confusion for users.

Reply:
You could use https://github.com/istio/istio/blob/405f47f54d19e6d998aa3e863ec9e2bc23c157f1/pkg/kube/config.go#L30 or https://github.com/istio/istio/blob/405f47f54d19e6d998aa3e863ec9e2bc23c157f1/pkg/kube/config.go#L54.